pilot/pkg/leaderelection/leaderelection.go

```diff
@@ -142,8 +153,13 @@ func (l *LeaderElection) create() (*k8sleaderelection.LeaderElector, error) {
 	if l.prioritized {
 		// Function to use to decide whether this revision should steal the existing lock.
 		config.KeyComparison = func(currentLeaderRevision string) bool {
+			var currentLeaderRemote bool
+			if currentLeaderRemote = strings.HasPrefix(currentLeaderRevision, remoteIstiodPrefix); currentLeaderRemote {
+				currentLeaderRevision = strings.TrimPrefix(currentLeaderRevision, remoteIstiodPrefix)
+			}
+			preferredLocation := !l.remote && currentLeaderRemote // this revison is running locally while the current one is remote
 			defaultRevision := l.defaultWatcher.GetDefault()
-			return l.revision != currentLeaderRevision &&
+			return (l.revision != currentLeaderRevision || preferredLocation) &&
```

Comment:
confused by this. this means we will only steal it if we are preferredLocation AND the default revision. But what if the local and remote are both `rev=not-default`. Don't we still want stealing?

Reply:
The current behavior is that the only way the lock is ever stolen is if the new one is a different revision and it's the default revision. I'm adding one more condition to also allow a local default revision to steal from a remote default revision. Non-default revisions will never steel the lock, before or after this change.

If we want to allow non-default local revisions to steal locks from non-default remote revisions, I think that would be fine, but it would need to also check that the local and remote are the same non-default revision, or we will end up with trashing.

I can add that change, if you'd like. This change is really just an optimization to try to have the local one be the preferred controller, but things still work without it, so that's why I was trying to change the behavior as little as possible for now, thinking we can always further optimize later, but I'm fine with adding the non-default revision remote->local override as well. Let me know what you think.